crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -546,6 +546,44 @@ impl Mesh {
         self
     }
 
+    /// Inverts the winding of the indices such that all counter-clockwise triangles are now
+    /// clockwise and vice versa.
+    ///
+    /// Does nothing if no [`Indices`] are set.
+    /// If this operation succeeded, an [`Ok`] result is returned.
+    pub fn invert_winding(&mut self) -> Result<(), MeshWindingInvertError> {
+        if self.primitive_topology != PrimitiveTopology::TriangleList {
+            return Err(MeshWindingInvertError::WrongTopology);
+        }
+
+        fn invert<'a, I: 'static>(
+            indices: impl Iterator<Item = &'a mut [I]>,
+        ) -> Result<(), MeshWindingInvertError> {
+            for chunk in indices {
+                // We check this every run because it will be an overhead as long as we
+                // don't use unsafe checks inside.
+                let [_, b, c] = chunk else {
+                    return Err(MeshWindingInvertError::AbruptIndicesEnd);
+                };
+                std::mem::swap(b, c);
+            }
+            Ok(())
+        }
+        match &mut self.indices {
+            Some(Indices::U16(vec)) => invert(vec.chunks_mut(3)),
+            Some(Indices::U32(vec)) => invert(vec.chunks_mut(3)),
+            None => Ok(()),
+        }
```

Comment:
```suggestion
        fn invert<I>(
            indices: &mut [I],
        ) -> Result<(), MeshWindingInvertError> {
            // Early return if the index count doesn't match
            if indices.len() % 3 != 0 {
                return Err(MeshWindingInvertError::AbruptIndicesEnd);
            }
            for chunk in indices.chunks_mut(3) {
                // We check this every run because it will be an overhead as long as we
                // don't use unsafe checks inside.
                let [_, b, c] = chunk else {
                    return Err(MeshWindingInvertError::AbruptIndicesEnd);
                };
                std::mem::swap(b, c);
            }
            Ok(())
        }
        match &mut self.indices {
            Some(Indices::U16(vec)) => invert(vec),
            Some(Indices::U32(vec)) => invert(vec),
            None => Ok(()),
        }
```
I feel like it might still be desirable to do a quick check that `indices.len()%3 == 0` at the start of `fn invert`, just for the sake of not leaving the mesh partially flipped, and skipping that work in case of an error.

Reply:
Ill have this in mind, thanks! Didn't thought about the mesh after an error.
Ill se if I can somehow get the double check removed without unsafe, maybe with unreachable!